cmd/snap-confine/snap-confine-invocation.h

```diff
@@ -31,6 +31,7 @@ typedef struct sc_invocation {
     /* Things declared by the system. */
     char *snap_instance;
     char *base_snap_name;
+    char *fallback_base_snap_name;
```

Comment:
let's simplify this further,  have both:

*  `base_snap_name`   this is used almost everywhere as before
* `original_base_snap_name`    this should be something only some of the code in main should care about

always have both set, normally they have the same value,  if there's a fallback base_snap_name is set to it.

I'm also ok to call `base_snap_name`  `effective_base_snap_name` if preferred.

Reply:
Done, I changed my mind and used your suggestion directly.